pkg/packageinstall/reconciler.go

```diff
@@ -78,5 +80,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 
-	return NewPackageInstallCR(existingPkgInstall, log, r.kcClient, r.pkgClient, r.coreClient).Reconcile()
+	pkgi := NewPackageInstallCR(existingPkgInstall, log, r.kcClient, r.pkgClient, r.coreClient)
+	if err := pkgi.Validate(); err != nil {
+		pkgi.model.Status.FriendlyDescription = "Invalid"
+		pkgi.model.Status.UsefulErrorMessage = err.Error()
+		pkgi.model.Status.Conditions = []v1alpha1.AppCondition{{
+			Type:   kappctrlv1alpha1.ReconcileFailed,
+			Status: corev1.ConditionTrue,
+		}}
+		if err := pkgi.UpdateStatus(); err != nil {
+			return reconcile.Result{Requeue: true}, err
+		}
```

Comment:
instead of modifying private model on pkgi and exposing UpdateStatus(), i think it would be better to make a new public method on PackageInstallCR that modifies state and updates status e.g.

```
func (pi *PackageInstallCR) UpdateAsReconcileFailed(err error) error
```

wdyt?

Reply:
Not crazy about that idea, `UpdateAsReconcileFailed` seems an overly specific method. I was trying to have the `Validate` method be more of a pure data model type method, but right now the `PackageInstall` is more of a delegated reconciler/controller object, with it doing all the client creates/updates. 

I think I see two options:
- move validate to be called by pkgi.Reconcile
- move UpdateStatus to outside Reconcile and have Validate update status

I think I prefer the first one just because it's more in line with the rest of the reconcilers and I don't want to boil the ocean, but I think I prefer the second one in the larger scheme of things, it reminds me that in Knative their reconciler framework automatically updates status after objects are reconciled